### PR TITLE
Remove site style step from flows and update default theme to Maywood

### PR DESCRIPTION
**Pull request by andrewserong**
#### Changes proposed in this Pull Request

* Update default theme for `blog`, `business`, and `professional` segments to use `pub/maywood`
* Remove `site-style-with-preview` step from flows
* Remove site style step from e2e tests
* Remove the site style import to reduce bundle size, and instead use a hard-coded font URL
* Include the Gutenberg `theme.css` file, which includes overrides for separators
* Add a couple of styling overrides to ensure site preview matches newly created sites using the Maywood theme

#### Testing instructions

* Apply D32188-code and D32317-code
* Complete signup flows from `http://calypso.localhost:3000/start` and ensure that the preview looks like the screenshots below. Test on Chrome, Firefox, Safari, and IE11
* Ensure that all flows where the `site-style-with-preview` step has been removed are still functioning correctly (`business`, `premium`, `personal`, `free`, and `onboarding` flows)
* Ensure that the created site is using the Maywood theme
* Create an e-commerce site, and ensure that it is still using the `pub/dara` theme

Screenshots of the site preview and newly created sites for each of the affected site types (segments):

## Blog preview versus newly created site

image

## Business preview versus newly created site

maywood-business

## Professional preview versus newly created site

maywood-professional

Related to issue #35842

### CODE REVIEW

**Comment by andrewserong** [comment]
## Update on styling inconsistencies

There are quite a few little styling fixes to be done. I've audited the differences between the site preview and created sites across the three segments `blog`, `professional`, and `business`. The following need to be fixed (the solution might be a combination of some CSS overrides and some slight adjustments to markup):

### Blog site

* [ ] Site title is lighter font-weight in site preview than in created site (300 instead of 700)
* [ ] Top and bottom margins on `.a8c-posts-list__item` are missing, meaning that the `Featured` label, title, and date are all too close together in the posts list

```
.a8c-posts-list__item .entry>* {
    margin-top: 16px;
    margin-bottom: 16px;
}
```

* [ ] HR separator has a slight border, instead of being solid grey
* [ ] Subscribe button layout looks quite broken (I think this one can be fixed by swapping the markup from a `<span>` to a `<button>`)

### Business site

* [ ] Site title is lighter font-weight in site preview than in created site (300 instead of 700)
* [ ] HR separator is full-width and has a slight border, instead of being narrow and a solid grey

### Professional site

* [ ] Site title is the wrong font-weight
* [ ] Width and styling of HR separators
* [ ] Client logos are too big on mobile

**Comment by andrewserong** [comment]
This one's ready for review now. There are further changes to be made to the business vertical template, but they're to be made on the backend, which I _think_ can be done separately to this PR (heading size and centre alignment mentioned in https://github.com/Automattic/wp-calypso/issues/35842#issuecomment-526346797 )

**Comment by andrewserong** [comment]
The e2e tests have been failing, but I _think_ these are unrelated failures in Gutenberg. I will double-check and take a closer look on Monday.

**Comment by ramonjd** [comment]
Checked in Chrome, FF, Safari, IE and Android. Looks good.

**Comment by p-jackson** [comment]
I noticed that the preview during signup said "Welcome to My New Cameras & Photography Blog" over the mountain vista. However on the created site it says "Welcome to My New Blog".

see this during signup

this when viewing my site

Maybe headstarting a site has always left this content as the default and I've only just noticed it.

Using onboarding flow, blog segment, site topic = Cameras & Photography

**Comment by andrewserong** [comment]
@p-jackson thanks for testing! I can replicate this on master and in this PR, but only with a site topic that has an ampersand in it. E.g. `Cameras and Photography` works fine, but `Cameras & Photography` results in an empty space where the placeholder would usually be. You can see the slight extra gap in the block editor:

image

I'm guessing that the string's not passing validation because it has an illegal character in it. Since there's a bit of time pressure on getting this PR merged early this week and the issue exists on master, I'm thinking we can fix this issue in a separate PR / diff.

**Review by spen — APPROVED** [review]
> Since there's a bit of time pressure on getting this PR merged early this week and the issue exists on master, I'm thinking we can fix this issue in a separate PR / diff.

+1 on that.

Looks to be working well for me, great work @andrewserong  :)

**Review by p-jackson — APPROVED** [review]
> the issue exists on master

Ah so it does! Changes look good to me.

**Comment by ramonjd** [comment]
> I'm guessing that the string's not passing validation because it has an illegal character in it. 

I'll take a look at this today to see if I can fix. You're right: defo not a blocker. 

Great work!

File: client/lib/signup/site-styles.js [hunk 1]
```diff
@@ -105,6 +105,13 @@ export const siteStyleOptions = {
 export const getThemeCssUri = ( themeSlug, isRtl ) =>
 	`https://s0.wp.com/wp-content/themes/${ themeSlug }/style${ isRtl ? '-rtl' : '' }.css`;
 
+/**
+ * Returns a default font URI
+ *
+ * @return {String}            The default font URI
+ */
+export const getFontUrl = () => 'https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,500,700';
```
[the remark this answers is not in the document]
Comment (andrewserong) [reply client/lib/signup/site-styles.js]: > Could this be a constant?

Good call! Thanks @spen :)

**Comment by ramonjd** [comment]
> I'm guessing that the string's not passing validation because it has an illegal character in it. Since there's a bit of time pressure on getting this PR merged early this week and the issue exists on master, I'm thinking we can fix this issue in a separate PR / diff.

Just to follow up on this. 

It's because Headstart doesn't know about non-"official" verticals. Only vertical ids, and will hence return `''` if it can't find a vertical among our list.

We could implement the same restriction on the frontend, by updating the [vertical name prop we pass to the site preview component](https://github.com/Automattic/wp-calypso/blob/a7efcec/client/signup/site-mockup/index.jsx#L213), e.g.,

```
                 ...
		const siteVerticalName = getSiteVerticalIsUserInput( state ) ? '' : getSiteVerticalName( state );

		return {
			...
			siteVerticalName,
                         ...
```

Or we could try to update Headstart to accept and handle an arbitrary vertical name (user input).

The latter might take a while.

It doesn't appear to be a huge blocker, so we can park it and discuss.

**Pull request merged**